fasthtml/js.py

```diff
@@ -87,3 +88,47 @@ def SortableJS(
 proc_htmx('%s', Sortable.create);
 """ % sel
     return Script(src, type='module')
+
+# %% ../nbs/api/03_js.ipynb
+def MermaidJS(
+        sel='.language-mermaid',  # CSS selector for mermaid elements
+        theme='base',  # Mermaid theme to use
+        delay=500  # Delay in milliseconds before rendering
+    ):
+    "Implements browser-based Mermaid diagram rendering."
+    src = """
+import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@11/dist/mermaid.esm.min.mjs';
+
+mermaid.initialize({
+    startOnLoad: false,
+    theme: '%s',
+    securityLevel: 'loose',
+    flowchart: { useMaxWidth: false, useMaxHeight: false }
+});
+
+function renderMermaidDiagrams() {
+    const diagrams = document.querySelectorAll('%s');
+    diagrams.forEach((element, index) => {
+        try {
+            const graphDefinition = element.textContent;
+            const graphId = `mermaid-diagram-${index}`;
+            mermaid.render(graphId, graphDefinition)
+                .then(({svg, bindFunctions}) => {
+                    element.innerHTML = svg;
+                    bindFunctions?.(element);
+                })
+                .catch(error => {
+                    console.error(`Error rendering Mermaid diagram ${index}:`, error);
+                    element.innerHTML = `<p>Error rendering diagram: ${error.message}</p>`;
+                });
+        } catch (error) {
+            console.error(`Error processing Mermaid diagram ${index}:`, error);
+        }
+    });
+}
+
+proc_htmx('%s', () => {
+    setTimeout(renderMermaidDiagrams, %d);
```

Comment:
Why the delay? I removed it from the code you submitted for my site, and it [appears to work fine](https://daniel.feldroy.com/posts/mermaid-charts). I could be in error, if so we'll keep it in.

Reply:
Kept it to say to see the change happening , removed it now, should be working fine without that as well, thanks

Reply:
Afaict this isn't using proc_htmx correctly (although I might be missing something) - have a look at the proc_htmx source and make sure you understand its purpose and use.

Reply:
Thanks for the feedback @jph00  , proc_htmx already iterates with a selector and a js function. Changed it , code looks much more compact now.